Mathlib/Order/Defs/Unbundled.lean

```diff
@@ -272,6 +272,34 @@ structure LowerSet (α : Type*) [LE α] where
 
 extend_docs LowerSet before "The type of lower sets of an order."
 
+/-- An upper set relative to an element `c` is a set such that all elements are at most `c` and
+any element greater than one of its members and less than `c` is also a member. -/
+def IsRelUpperSet {α : Type*} [LE α] (s : Set α) (c : α) : Prop :=
+  ∀ ⦃a : α⦄, a ∈ s → a ≤ c ∧ ∀ ⦃b : α⦄, a ≤ b → b ≤ c → b ∈ s
+
+/-- A lower set relative to an element `c` is a set such that all elements are at least `c` and
+any element less than one of its members and greater than `c` is also a member. -/
+def IsRelLowerSet {α : Type*} [LE α] (s : Set α) (c : α) : Prop :=
+  ∀ ⦃a : α⦄, a ∈ s → c ≤ a ∧ ∀ ⦃b : α⦄, b ≤ a → c ≤ b → b ∈ s
+
+@[inherit_doc IsRelUpperSet]
+structure RelUpperSet {α : Type*} [LE α] (c : α) where
+  /-- The carrier of a `RelUpperSet`. -/
+  carrier : Set α
+  /-- The carrier of a `RelUpperSet` is an upper set relative to `c`. -/
+  relUpper : IsRelUpperSet carrier c
```

Comment:
```suggestion
  isRelUpperSet' : IsRelUpperSet carrier c
```
and please add `isRelUpperSet` in terms of `SetLike.coe`. Same below

Reply:
Done.